src/main/scala/cromwell/engine/io/gcs/GoogleCloudStorage.scala

```diff
@@ -103,9 +102,14 @@ case class GoogleCloudStorage private(client: Storage) extends IoInterface {
   def writeFile(path: String, content: String): Unit = uploadObject(path, content)
 
   def writeTempFile(path: String, prefix: String, suffix: String, content: String): String = {
-    val fullPath = s"$path/$prefix${UUID.randomUUID()}$suffix"
-    writeFile(fullPath, content)
-    path
+    // This may be called multiple times with the same inputs.  Calling this twice with the same
+    // parameters should yield the same return value.
+    val hash = content.md5Sum
+    val fullPath = s"$path/$prefix${hash.substring(0, 8)}$suffix"
```

Comment:
This was related to a bug fix.  Consider this WDL

```
task x {
  Array[String] arr
  command {
    ./script --file=${write_lines(arr)}
  }
}
```

when we evaluate this command, `writeTempFile` will be called to write the results of `write_lines(arr)`.

BUT, for JES, we need to tell JES *ahead of time* what this file is and where to localize it.  So in the `JesBackend.generateJesInputs` ([here](https://github.com/broadinstitute/cromwell/pull/470/files#diff-bbfaf645248cd47b15a6322ebf4ccbc1R435)), we run the command through the `FileEvaluator`, which will return a `Seq[WdlFile]` for all files referenced in the command.  The `FileEvaluator` will evaluate `write_lines(arr)`, write the temp file, and return the path so we can include it in the `JesInput` list.

THEN, later on, we'll instantiate the command again.  Both times we evaluate `write_lines(arr)` and both times it needs to return the SAME value.

I realize this is a little confusing, so any suggestions to help make it cleaner would be greatly appreciated